### ISSUE TITLE
QScrollArea requires two taps to trigger click event on elements contained within [iOS]

### ISSUE THREAD

**Issue by thexeos** [quasarframework/quasar#16210]
### What happened?

The way Safari was programmed to work, means that having a @mouseenter event that modifies DOM prevents event like @mouseup and @click from being triggered down the DOM tree. In case of QScrollArea, @mouseenter event is used to display the scroll thumb, which means that a second tap is required to interact with elements within QScrollArea.

![image](https://github.com/quasarframework/quasar/assets/2260539/cb738e8e-9f4c-4e60-aabd-af35abc87a40)

Taken from [this Safari documentation page](https://developer.apple.com/library/archive/documentation/AppleApplications/Reference/SafariWebContent/HandlingEvents/HandlingEvents.html)

### What did you expect to happen?

Now, Apple would not be Apple if it was consistent with its issues. This does not happen on some devices and happens on others, but I was able to successfully reproduce this with BrowserStack. I checked WebKit code, but I couldn't find any specific reference to the implementation of this behavior.

Quasar code that triggers this Safari issue is this one:

https://github.com/quasarframework/quasar/blob/f8b458292ec1caef77420cc5197454ee66fe4fc8/ui/src/components/scroll-area/QScrollArea.js#L355-L357

### Reproduction URL

NA

### How to reproduce?

The workaround requires delaying any DOM modifications until a short cooldown period had passed. From my testing, we don't have to wait until mouseup event had triggered, but making modification within setTimeout(0), rAF() or nextTick() leaves insufficient delay. Making it a capture event doesn't make a difference either. The safest approach that I found is to do:

```
function onMouseenter () {
  setTimeout(() => {
    hover.value = true
  }, proxy.$q.platform.is.ios ? 50 : 0)
}
```

The 50ms delay is arbitrary, but it seems to work for me every time.

### Flavour

Quasar CLI with Vite (@quasar/cli | @quasar/app-vite)

### Areas

Components (quasar)

### Platforms/Browsers

iOS

### Additional context

Current behavior:

![opera_5dfOrPgr3E](https://github.com/quasarframework/quasar/assets/2260539/ee3c1b82-0c5f-4903-87cd-56e3e67b7162)

With the proposed change:

![opera_rXPMGQUO8D](https://github.com/quasarframework/quasar/assets/2260539/fcc23e9e-7012-407f-aab9-36b42ca2baf2)

**Comment by HacMan137** [comment]
Thank you for making an issue and MR for this. I have the same issue in my project. Hoping your MR gets merged soon. Out of curiosity, do you know of any work-arounds that could be used to fix this issue in the short term that don't involve patching the quasar source?

**Comment by thexeos** [comment]
@HacMan137 I don't know of any other way other than applying the patch to your local copy of Quasar in node_modules.

**Comment by rstoenescu** [comment]
Fix will be available in Quasar v2.12.8.
Thank you for the detailed explanation and also for providing a PR!
